Added menu bar in MainForm.

mllrdev-carlo-medel opened the pull request:
This commit will also make the adding
of items to be independent compared to
the previous implementation. Also removed
the generation of primary key from the
application layer, it is now handled by the
database.

mllrdev-carlo-medel committed:
Added menu bar in MainForm.

This commit will also make the adding
of items to be independent compared to
the previous implementation. Also removed
the generation of primary key from the
application layer, it is now handled by the
database.

KarlZablan commented on Forms/AddItemsForm.cs:
```diff
@@ -0,0 +1,194 @@
+            }
+        }
+
+        private void SearchButton_Click(object sender, EventArgs e)
```
Rework. Why should I fetch all records to search?

KarlZablan commented on Forms/CartForm.cs:
```diff
@@ -195,8 +195,8 @@ private void RemoveButton_Click(object sender, EventArgs e)
                                     itemsForm.LoadData();
                                 }
 
```
Rework

KarlZablan commented on Forms/ItemsForm.cs:
```diff
@@ -42,67 +42,24 @@ public void LoadData()
                 foreach (Item item in _itemManager.GetAll())
                 {
-                    ListViewItem listViewItem = new ListViewItem(new[] {string.Empty,
+                    if (item.Stocks > 0)
```
Same comment with AddItemForm's Search functionality

KarlZablan commented on Forms/ProfileForm.cs:
```diff
@@ -58,11 +58,21 @@ public void LoadData()
 
                 foreach (PurchaseHistory purchaseHistory in _customer.PurchaseHistory.OrderByDescending(x => DateTime.Parse(x.Purchase.Date)))
                 {
+                    if (purchaseHistory.Purchase.Status == ProfileStringConstants.PENDING.FirstCharToUpper())
```
don't manipulate constants

KarlZablan reviewed (changes requested):
Please Check changes

mllrdev-carlo-medel committed:
Bug fix for updating items

When updating the items, and if that item
belongs to the pending purchase then update it,
otherwise don't.

KarlZablan commented on Forms/AddItemsForm.cs:
```diff
@@ -189,5 +189,46 @@ private void AddItemsForm_FormClosing(object sender, System.ComponentModel.Cance
+                {
+                    using (TransactionScope scope = new TransactionScope())
+                    {
+                        List<Item> items = new List<Item>();
```
lines 201 - 206, please move it outside of the using statement.

KarlZablan commented on Forms/CartForm.cs:
```diff
@@ -246,6 +246,19 @@ private void CheckoutButton_Click(object sender, EventArgs e)
                         _purchase.Date = DateTime.Now.ToString();
                         _purchase.Total = ComputeTotalPrice();
 
+                        foreach (PurchaseDetails purchaseDetails in _purchases)
```
Optimize

mllrdev-carlo-medel replied:
Will remove this. This part was already handled in customer factory.

KarlZablan commented on Forms/ProfileForm.cs:
```diff
@@ -62,7 +62,7 @@ public void LoadData()
                     {
                         foreach (PurchaseDetails purchaseDetails in purchaseHistory.PurchaseDetails)
```
line 63 - 68: Please optimize

KarlZablan reviewed (changes requested):
Please check comments

mllrdev-carlo-medel committed:
Optimized the computation of total price for the purchase.
